04_numeronym/hsy3418/main.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, numeronyms("accessibility", "Kubernetes", "abc"))
+}
+
+//numeronyms implements a function that returns a slice of numeronyms for its input strings
+func numeronyms(vals ...string) []string {
+	numeronymsList := []string{}
+	for _, val := range vals {
+		countLetters := len(val)
+		if countLetters > 3 {
+			var buf strings.Builder
+			firstChar := val[:1]
+			lastChar := val[len(val)-1:]
+			midLength := strconv.Itoa(len(val) - 2)
+			buf.WriteString(firstChar)
+			buf.WriteString(midLength)
+			buf.WriteString(lastChar)
+			val = buf.String()
```

Comment:
Not essential but IMO it might be tidier if you build the new numeronym value in a single line... like this 
numeronymsList[i] = val[0] + strconv.Itoa(len(val)-2) + alphaNumChar[len(val)-1].  And if you just set your numeronymsList := vals at the beginning then you don't need to do anything when the len is less than 3.